cloud/amazon/ec2_asg.py

```diff
@@ -281,18 +300,12 @@ def create_autoscaling_group(connection, module):
 
     asg_tags = []
     for tag in set_tags:
-        if tag.has_key('key') and tag.has_key('value'):  # this block is to support depricated form
-            asg_tags.append(Tag(key=tag.get('key'),
-                value=tag.get('value'),
-                propagate_at_launch=bool(tag.get('propagate_at_launch', True)),
-                resource_id=group_name))
-        else:
```

Comment:
Yes it will break backwards compatibility. I didn't know how to test the legacy format and it was marked as "deprecated", so I stripped it.